compatibility_server/pip_checker.py

```diff
@@ -177,43 +202,100 @@ class _OneshotPipCheck():
             packages: The packages to check for compatibility e.g.
                 ['numpy', 'tensorflow'].
-            tmp_path: The file system path to use for temporary files
-                e.g."/tmp".
-            clean: If True then all previously installed packages will be
-                uninstalled before installing "packages".
         """
         self._pip_command = pip_command
         self._packages = packages
-        self._tmp_path = tmp_path
-        self._output_directory = None
-        self._clean = clean
 
-    @staticmethod
-    def _run_command(command, stdout_path, stderr_path, raise_on_failure=True):
-        with open(stdout_path, 'w') as stdout_file, open(stderr_path,
-                                                         'w') as stderr_file:
-            completed_command = subprocess.run(
-                command, stdout=stdout_file, stderr=stderr_file)
+    def _get_base_image(self):
+        """Get the base image name based on Python version."""
+        python_version = self._pip_command[0]
 
-        logging.debug('Running %s [returncode=%s]', command,
-                      completed_command.returncode)
-        if completed_command.returncode and raise_on_failure:
-            raise PipError(command, completed_command.returncode, stderr_path)
+        if python_version == 'python2':
+            base_image = "python:2.7"
+        else:
+            base_image = "python:3.6"
 
-        return completed_command.returncode
+        return base_image
+
+    def _build_container(self, docker_client):
+        """Build the container which contains a Python interpreter. The timeout
+        for running the commands is 300 seconds. Container will stop running
+        after the timeout.
+        """
+        base_image = self._get_base_image()
+
+        try:
+            container = docker_client.containers.run(
+                base_image,
+                command="sleep {}".format(TIME_OUT),
+                detach=True)
+        except docker.errors.APIError as e:
+            raise PipCheckerError(
+                error_msg="Error occurs when starting docker container."
+                          "Error message: {}".format(e.explanation))
+
+        return container
+
+    def _cleanup_container(self,
+                           container: docker.models.containers.Container):
+        """Remove the container."""
+        try:
+            container.stop()
+            container.remove()
+        except (docker.errors.APIError, docker.errors.NotFound):
+            raise PipCheckerError(
+                error_msg="Error occurs when cleaning up docker container."
+                          "Container does not exist.")
+
+    def _run_command(self,
+                     container: docker.models.containers.Container,
+                     command: List[str],
+                     stdout: bool,
+                     stderr: bool,
+                     raise_on_failure: Optional[bool] = True):
```

Comment:
You can add the return value as a type annotation.

Reply:
Done.